src/core/net/ip6.cpp

```diff
@@ -477,6 +479,8 @@ Error Ip6::SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, uint8_t aI
             if (messageCopy != nullptr)
             {
                 LogInfo("Message copy for indirect transmission to sleepy children");
+                messageCopy->SetAllowLoopbackToHost(true);
+                messageCopy->SetOrigin(Message::kOriginHostTrusted);
```

Comment:
Can we let the `Clone()` copy the `origin` and `AllowLoopback` valud into `messageCopy`? Do we need to set it?

Reply:
In this commit `Clone()` already copies the `mOrigin` and `mAllowLoopback`. However, there're sometimes we have to reset them probably because the former origin is `kOriginThreadNetif`. Otherwise some tests will fail, e.g. `Cert_9_2_12_Announce.py` and `Cert_9_2_05_PanIdQuery.py`.

If we want to call these two setters less, we may consider let `Clone()` not copy the `origin` and `allowloopback` but set them to the default value (`kOriginHostTrusted`, `true`). What do you think?

Reply:
I think `Clone()` copying the origin and other flags make sense.

I think it may be good to figure out why/how test are failing. This is in `SendDatagram()`. May be good to check where would we call this with a `message` with `kOriginThread` (so to make sure that we may be missing setting the origin from somewhere?)

I checked all places we call `SendDatagram()`, as far as I can see all places should be a newly allocated message, so the origin should be already set to `host`?
```
~/github/openthread/src/core/meshcop/border_agent.cpp:
  337:     SuccessOrExit(error = Get<Ip6::Udp>().SendDatagram(*message, messageInfo, Ip6::kProtoUdp));

~/github/openthread/src/core/net/icmp6.cpp:
   76:     SuccessOrExit(error = Get<Ip6>().SendDatagram(aMessage, messageInfoLocal, kProtoIcmp6));
  122:     SuccessOrExit(error = Get<Ip6>().SendDatagram(*message, messageInfoLocal, kProtoIcmp6));
  214:     SuccessOrExit(error = Get<Ip6>().SendDatagram(*replyMessage, replyMessageInfo, kProtoIcmp6));

~/github/openthread/src/core/net/tcp6.cpp:
  985:     IgnoreError(instance.Get<ot::Ip6::Ip6>().SendDatagram(message, info, kProtoTcp));

~/github/openthread/src/core/net/udp6.cpp:
  358:         SuccessOrExit(error = SendDatagram(aMessage, messageInfoLocal, kProtoUdp));
  430: Error Udp::SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, uint8_t aIpProto)
  454:         error = Get<Ip6>().SendDatagram(aMessage, aMessageInfo, aIpProto);

~/github/openthread/src/core/net/udp6.hpp:
  594:     Error SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, uint8_t aIpProto);

~/github/openthread/src/core/thread/mle.cpp:
 3945:     SuccessOrExit(error = Get<Ip6::Ip6>().SendDatagram(*message, messageInfo, Ip6::kProtoNone));
```

Reply:
I see. I'll try to figure out why the tests failed.

Reply:
I removed the `SetOrigin` calls in `ip6.cpp`.